database/factories/PackageFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Database\Factories;
+
+use Illuminate\Database\Eloquent\Factories\Factory;
+use App\Models\Plan;
+use App\Models\Package;
+use App\Models\Service;
+
+/**
+ * @extends \Illuminate\Database\Eloquent\Factories\Factory<\App\Models\Package>
+ */
+class PackageFactory extends Factory
+{
+    /**
+     * Define the model's default state.
+     *
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        $plan = Plan::factory()->create();
+
+        return [
+            'name' => $this->faker->word,
+            'plan_id' => $plan->id,
+            'price' => $plan->price,
+        ];
+    }
+
+    public function withServices($numServices = 3)
+    {
+        return $this->afterCreating(function (Package $package) use ($numServices) {
+            $services = Service::inRandomOrder()->take($numServices)->get();
+            $package->services()->attach($services);
+        });
```

Comment:
This may not work because it requires the services to be created before
You can use `$this->hasAttached` instead and create trought factory